File: BI-TJV/semestral/rest-server/src/main/java/wrzecond/repository/TjvTableRepository.java
package wrzecond.repository;

import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;
import wrzecond.entity.TjvTable;
import wrzecond.table.TjvTableType;

import java.util.List;

@Repository
public interface TjvTableRepository extends TjvRepository<TjvTable> {

    @Query("SELECT t FROM TjvTable t WHERE t.type = :type")
    List<TjvTable> findByType (TjvTableType type);

}

File: BI-KOT/semestral/rest-server/src/main/kotlin/wrzecond/repository/TjvTableRepository.kt
package wrzecond.repository

import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import wrzecond.entity.TjvTable
import wrzecond.TjvTableType

@Repository
interface TjvTableRepository : TjvRepository<TjvTable> {

    @Query("SELECT t FROM TjvTable t WHERE t.type = :type")
    fun findByType (type: TjvTableType) : List<TjvTable>

}
